### ISSUE TITLE
How to suppress mobile browser keyboard

### ISSUE THREAD

**Issue by bizcad** [Mottie/Keyboard#204]
When applying Mottie Keyboard to a mobile device, on Android at least, a click on the text input brings up the Mottie Keyboard, but also the Android keyboard.  I want to not show the android keyboard.  I have tried the blur hack <a href="http://www.sencha.com/forum/archive/index.php/t-141560.html?s=6853848ccba8060cc90a9cacf7f0cb44"> as discussed here</a> but it does not work.

Following is my ko.bindingHandler init where the keyboard is bound to the text element.  I am trying to just show a number pad so I filter viewModel items not decimal or integer.

<pre>
window.ko.bindingHandlers.SetInputType = {
init: function (element, valueAccessor, allBindingsAccessor, viewModel) {        
        // the viewModel has a data variable called dataColumnType which indicates
        // what type of input element it is.  The if statement does not bind
        // mottie to text fields.
        if (viewModel.dataColumnType() === 'decimal' || viewModel.dataColumnType() === 'integer') {
            //    var marker = $('<span />').insertBefore($el);
            //    $($el).detach().attr('type', 'number').insertAfter(marker);
            //    marker.remove();
            if (!isNaN(app.vm.reading.reading().value())) {
                $(element).filter(':input:focus').blur();
                $(element).keyboard({
                    layout: 'custom',
                    display: {
                        'deg': 'D:degrees', // degrees
                        'rad': 'R:radians', // radians
                        'grad': 'G:grads', // grads
                        '10x': '10\u207f',
                        '2pi': '2\u03c0',
                        'acos': 'cos\u207b\u00b9',
                        'acosh': 'cosh\u207b\u00b9',
                        'asin': 'sin\u207b\u00b9',
                        'asinh': 'sinh\u207b\u00b9',
                        'atan': 'tan\u207b\u00b9',
                        'atanh': 'tanh\u207b\u00b9',
                        'clear': 'CE',
                        'clearall': 'C',
                        'cuberoot': '\u221bx',
                        'equals': '=',
                        'invx': '1\u2044x', // replaced with an image
                        'm-': 'M-',
                        'mc': 'MC',
                        'meta1': 'Inv', // show inverse trig function keyset
                        'mplus': 'M+',
                        'mr': 'MR',
                        'ms': 'MS',
                        'pi': '\u03c0',
                        'sqrt': '\u221a',
                        'x2': 'x\u00b2',
                        'x3': 'x\u00b3',
                        'xy': 'xy',
                        'yroot': 'yroot' // replaced with an image
                    },
                    customLayout: {
                        'default': [                            
                        ' ( ) {sign} {b}',
                        ' 7 8 9 /',
                        ' 4 5 6 *',
                        ' 1 2 3 -',
                        ' 0 {dec} +',
                        ' {c} {a}  {equals} {clear}'
                        ],
                        'meta1': [
                        ' ( ) {sign} {b}',
                        ' 7 8 9 /',
                        ' 4 5 6 *',
                        ' 1 2 3 -',
                        ' 0 {dec} +',
                        ' {c} {a}  {equals} {clear}'
                        ]
                    },

                    // Turning restrictInput on (true), prevents yroot and but it
                    restrictInput: true,  // Prevent keys not in the displayed keyboard from being typed in
                    useCombos: false, // don't want A+E to become a ligature
                    wheelMessage: '',    // clear tooltips

                    // set up degree/radian/grads mode
                    visible: function (e, kb, el) { // e = event, kb = keyboard object, el = original input
                        var mode = kb.mode || 0, sel = '.ui-keyboard-';
                        sel += (mode === 1) ? 'deg' : (mode === 2) ? 'grad' : 'rad';
                        kb.$keyboard.find(sel).addClass(kb.options.css.buttonAction);
                    },

                    // multiple parameter functions highlight when first parameter is saved,
                    // this makes sure the buttons aren't highlighted when the memory storage clears.
                    change: function (e, kb, el) {
                        if (kb.memory2 === '') {
                            kb.$keyboard.find('.ui-keyboard-xy, .ui-keyboard-yroot').removeClass(kb.options.css.buttonAction);
                        }
                    }

                }).addTyping();

                // the blur hack.
                element.setAttribute('readonly', 'readonly'); // Force keyboard to hide on input field.
                element.setAttribute('disabled', 'true'); // Force keyboard to hide on textarea field.
                setTimeout(function () {
                    element.blur();
                    // Remove readonly attribute after keyboard is hidden.
                    element.removeAttribute('readonly');
                    element.removeAttribute('disabled');
                }, 100);
            }
        }
        $(element).change(function () {
            var val = element.value;
            // posts the typed value back to the viewModel
            window.app.dataservice.setReadingValueForMRDId(viewModel, val);            
        });

        // dynamically add an id to the element for later reference.
        element.id = 'readinginput' + app.vm.reading.reading().id();

    },
    update: function (element, valueAccessor, allBindingsAccessor, viewModel) {
      // does nothing.
    }
};
</pre>

**Comment by ayushranjan-planetria** [comment]
Hi @Mottie , I am unable to suppress the native keyboard on iPad, I have tried with ```lockInput:true```, but it doesn't work. I tested the demo you shared "http://jsbin.com/eniQaBO/3/" on my iPad but the native keyboard also appears along with the mottie keyboard.
I also tried with ``` readonly="readonly" ``` and hacked the reveal() method by commenting these lines
```
/*if (base.$el.is(':disabled') || (base.$el.attr('readonly') &&
			!base.$el.hasClass(kbcss.locked))) {
			base.$el.addClass(kbcss.noKeyboard);
			return;
		} else {
			base.$el.removeClass(kbcss.noKeyboard);
		}*/
```
this works for the first time but as we start pressing the keyboard buttons, the iPad keyboard appears from nowhere. Can you please look into it?
Tested on iPad 2 (ios v8.3)
